kclvm/sema/src/advanced_resolver/node.rs

```diff
@@ -1235,32 +1250,62 @@ impl<'ctx> AdvancedResolver<'ctx> {
         args: &'ctx [ast::NodeRef<ast::Expr>],
         kwargs: &'ctx [ast::NodeRef<ast::Keyword>],
     ) -> anyhow::Result<()> {
-        for arg in args.iter() {
-            self.expr(arg)?;
-        }
+        self.do_arguments_symbol_resolve_with_hint(args, kwargs, false, vec![])
+    }
+
+    pub fn do_arguments_symbol_resolve_with_hint(
+        &mut self,
+        args: &'ctx [ast::NodeRef<ast::Expr>],
+        kwargs: &'ctx [ast::NodeRef<ast::Keyword>],
+        with_hint: bool,
+        mut params: Vec<Parameter>,
+    ) -> anyhow::Result<()> {
         for kw in kwargs.iter() {
             if let Some(value) = &kw.node.value {
                 self.expr(value)?;
             }
+            let kw_name = kw.node.arg.node.get_name();
             let (start_pos, end_pos): Range = kw.get_span_pos();
             let value = self.gs.get_symbols_mut().alloc_value_symbol(
-                ValueSymbol::new(kw.node.arg.node.get_name(), start_pos, end_pos, None, false),
+                ValueSymbol::new(kw_name.clone(), start_pos, end_pos, None, false),
                 self.ctx.get_node_key(&kw.id),
                 self.ctx.current_pkgpath.clone().unwrap(),
             );
 
             if let Some(value) = self.gs.get_symbols_mut().values.get_mut(value.get_id()) {
-                value.sema_info = SymbolSemanticInfo {
-                    ty: self
-                        .ctx
-                        .node_ty_map
-                        .borrow()
-                        .get(&self.ctx.get_node_key(&kw.id))
-                        .map(|ty| ty.clone()),
-                    doc: None,
-                };
+                let ty = self
+                    .ctx
+                    .node_ty_map
+                    .borrow()
+                    .get(&self.ctx.get_node_key(&kw.id))
+                    .map(|ty| ty.clone());
+                params.retain(|param| param.name != kw_name);
```

Comment:
if kwarg param is not removed from params, the hints will be like 
```
func = lambda x, y, z{
      x * y + z
}

z = func(x: a, y = 2, y: b)
```
here, y is displayed in inlay hints instead of z